### PR TITLE
Cleanup CentOS 7

### PR DESCRIPTION
**Pull request by braydonk**
## Description
CentOS 7 is now EOL, and all references and configuration for it are being removed.

## Related issue
[b/350698170](http://b/350698170)

## How has this been tested?
Integration tests in this PR.

## Checklist:
- Unit tests
  - [x] Unit tests do not apply.
  - [ ] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [ ] Integration tests do not apply.
  - [x] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [x] This PR introduces no user visible changes.
  - [ ] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [x] This PR introduces no new features.
  - [ ] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -272,16 +166,16 @@ RUN ./pkg/rpm/build.sh || true &> /dev/null
...
+COPY --from=rockylinux8-build-fluent-bit /work/cache /work/cache
+COPY --from=rockylinux8-build-systemd /work/cache /work/cache
+COPY --from=rockylinux8-build-diagnostics /work/cache /work/cache
+COPY --from=rockylinux8-build-wrapper /work/cache /work/cache
 RUN ./pkg/rpm/build.sh
 
-FROM scratch AS centos8
-COPY --from=centos8-build /tmp/google-cloud-ops-agent.tgz /google-cloud-ops-agent-centos-8.tgz
-COPY --from=centos8-build /google-cloud-ops-agent*.rpm /
+FROM scratch AS rockylinux8
```
Comment (igorpeshansky) [inline Dockerfile:176]: Careful. The name `centos8` is used by Rapid builds to refer the build, so I would suggest doing this in a staged manner — (1) rename `centos8-*`->`rockylinux8-*` everywhere, but keep `centos8` as an alias for `rockylinux8`, (2) switch the Rapid project to use `rockylinux8`, (3) remove the alias. This is better as a separate set of steps, so I would suggest not folding it into this PR.

File: kokoro/config/build/presubmit/rockylinux8_x86_64.gcl
Comment (igorpeshansky) [inline kokoro/config/build/presubmit/rockylinux8_x86_64.gcl:1]: We also have various Kokoro _job_ configs that reference `centos8`, so we would probably need this as a separate process. Let's pull this out into a separate PR.

**Review by igorpeshansky — COMMENTED** [review]
TL;DR: as-is, this PR will break our build infrastructure. Let's deal with `centos8`->`rockylinux8` in a separate PR.

**Comment by braydonk** [comment]
I've put all the CentOS 8 changes back where they were.

File: dev-docs/dev.md [hunk 2]
```diff
@@ -208,8 +208,6 @@ total 180116
 -rw-r--r-- 1 421646 89939  2919650 Oct  7 21:18 google-cloud-ops-agent_0.1.0~debian9.13_amd64.deb
 -rw-r--r-- 1 421646 89939  2914160 Oct  7 21:18 google-cloud-ops-agent_0.1.0~ubuntu18.04_amd64.deb
 -rw-r--r-- 1 421646 89939  2944116 Oct  7 21:18 google-cloud-ops-agent_0.1.0~ubuntu20.04_amd64.deb
--rw-r--r-- 1 421646 89939  7561251 Oct  7 21:16 google-cloud-ops-agent-centos-7.tgz
--rw-r--r-- 1 421646 89939  7726721 Oct  7 21:16 google-cloud-ops-agent-centos-8.tgz
```
[the remark this answers is not in the document]
Comment (braydonk) [reply dev-docs/dev.md:212]: [c073e73](https://github.com/GoogleCloudPlatform/ops-agent/pull/1753/commits/c073e736320b28b65107ecd3ddc5978710103eee)

File: integration_test/metadata/integration_metadata.go [hunk 3]
```diff
@@ -50,7 +50,7 @@ type ExpectedMetric struct {
 	// Exclusive metric to a particular kind of platform.
 	Platform string `yaml:"platform,omitempty" validate:"excluded_with=Representative,omitempty,oneof=linux windows"`
 	// A list of platforms that this metric is not available on.
-	// Examples: centos-7,debian-10. Not valid are linux,windows.
+	// Examples: debian-11. Not valid are linux,windows.
```
[the remark this answers is not in the document]
Comment (braydonk) [reply integration_test/metadata/integration_metadata.go:53]: [c073e73](https://github.com/GoogleCloudPlatform/ops-agent/pull/1753/commits/c073e736320b28b65107ecd3ddc5978710103eee)

File: kokoro/config/test/third_party_apps/release/centos8_x86_64.gcl
[the remark this answers is not in the document]
Comment (braydonk) [reply kokoro/config/test/third_party_apps/release/centos8_x86_64.gcl:1]: [c073e73](https://github.com/GoogleCloudPlatform/ops-agent/pull/1753/commits/c073e736320b28b65107ecd3ddc5978710103eee)

**Review by igorpeshansky — APPROVED** [review]
LGTM :shipit:

File: integration_test/README.md [hunk 4]
```diff
@@ -242,6 +242,8 @@ the corresponding `main_log.txt` from there.
 
 Here is the full contents uploaded to the GCS bucket for a single test run.
 The "Details" link takes you directly to the "logs" subdirectory to save you
+a hop. The following is sorted roughly in descending order of usefulness.
```
Comment (igorpeshansky) [inline integration_test/README.md:245]: The files below are sorted alphabetically, so maybe not worth mentioning the sort order…
```suggestion
a hop.
```

File: integration_test/README.md [hunk 5]
```diff
@@ -261,7 +263,6 @@ The "Details" link takes you directly to the "logs" subdirectory to save you
         ├── syslog.txt
         └── systemctl_status_for_ops_agent.txt
 ```
-
```
Comment (igorpeshansky) [inline integration_test/README.md:264]: [Optional] I'd keep this blank line…

Comment (braydonk) [reply integration_test/README.md]: Doesn't change the Markdown result, but I did it anyway in [46db9df](https://github.com/GoogleCloudPlatform/ops-agent/pull/1753/commits/46db9dfee78ae8468c868e5f14699160b2b79c78)

File: integration_test/README.md (hunk 4 above)
Comment (braydonk) [reply integration_test/README.md]: Done in [46db9df](https://github.com/GoogleCloudPlatform/ops-agent/pull/1753/commits/46db9dfee78ae8468c868e5f14699160b2b79c78)
